### ISSUE TITLE
Duplicate keys raise an error

### ISSUE THREAD

**Issue by wjandrea** [circlecell/jsonlint#1]
Example:
```json
{"name": "Alice", "age": 30, "age": 30}
```

JSONLint says:
> **Invalid JSON!**
> Error: Duplicate key 'age'

However, duplicate keys are valid. [ECMA-404](https://ecma-international.org/wp-content/uploads/ECMA-404_2nd_edition_december_2017.pdf) says:
> The JSON syntax does not impose any restrictions on the strings used as names, does not require that name strings be unique, ...

So this data is valid.

That said, I agree that duplicate keys aren't good. They should raise a warning instead.

**Comment by toddynho** [comment]
good call, we now show a warning
